sidekick_core/test/update_command_test.dart

```diff
@@ -0,0 +1,130 @@
+    final spec = ZoneSpecification(
+      print: (_, __, ___, line) => printLog.add(line),
+    );
+    await Zone.current.fork(specification: spec).run(() async {
+      await insideFakeProjectWithSidekick((projectDir) async {
+        final sidekickDir = projectDir.directory('packages/dash');
+        final expectedFilesToGenerate = [
+          'tool/download_dart.sh',
+          'tool/install.sh',
+          'tool/run.sh',
+          'tool/sidekick_config.sh',
+        ].map(sidekickDir.file);
+
+        for (final file in expectedFilesToGenerate) {
+          expect(
+            !file.existsSync() || file.readAsStringSync().isEmpty,
+            isTrue,
+            reason: '${file.path} exists or is not empty',
+          );
+        }
+
+        final runner = initializeSidekick(
+          name: 'dash',
+          dartSdkPath: systemDartSdkPath(),
+        );
+
+        runner.addCommand(UpdateCommand());
+        await runner.run(['update']);
+
+        const versionChecker = SidekickVersionChecker();
+
+        final sidekickVersionAfterUpdate = versionChecker
+            .getCurrentMinimumPackageVersion(['sidekick', 'generator_version']);
+        final sidekickCoreVersionAfterUpdate = versionChecker
+            .getCurrentMinimumPackageVersion(['dependencies', 'sidekick_core']);
+        final latestSidekickVersion =
+            await versionChecker.getLatestPackageVersion('sidekick');
+        final latestSidekickCoreVersion =
+            await versionChecker.getLatestPackageVersion('sidekick_core');
+
+        expect(sidekickVersionAfterUpdate, latestSidekickVersion);
+        expect(sidekickCoreVersionAfterUpdate, latestSidekickCoreVersion);
+
+        for (final file in expectedFilesToGenerate) {
+          expect(
+            file.existsSync() && file.readAsStringSync().isNotEmpty,
+            isTrue,
+            reason: '${file.path} does not exist or is empty',
+          );
+        }
+
+        expect(
+          printLog,
+          containsAllInOrder([
+            grey(
+              'Updating sidekick CLI dash from version 0.0.0 to $latestSidekickVersion ...',
+            ),
+            green(
+              'Successfully updated sidekick CLI dash from version 0.0.0 to $latestSidekickVersion!',
+            ),
+          ]),
+        );
+      });
+    });
+  });
+}
+
+R insideFakeProjectWithSidekick<R>(R Function(Directory projectDir) block) {
+  final tempDir = Directory.systemTemp.createTempSync();
+  'git init ${tempDir.path}'.run;
+
+  tempDir.file('pubspec.yaml')
+    ..createSync()
+    ..writeAsStringSync('name: main_project\n');
+  tempDir.file('dash').createSync();
+
+  final fakeSidekickDir = tempDir.directory('packages/dash')
+    ..createSync(recursive: true);
+
+  fakeSidekickDir.file('pubspec.yaml')
+    ..createSync()
+    ..writeAsStringSync('''
+name: dash
+
+environment:
+  sdk: '>=2.14.0 <3.0.0'
+
+dependencies:
+  sidekick_core: 0.0.0
+
+sidekick:
+  generator_version: 0.0.0
```

Comment:
We don't need `generator_version` to be a semver, a simple int is sufficient

Reply:
I read the latest version from pub.dev which is a semver and very easy compared to reading the latest version from the right file in the github repository at the right revision